compiler/passes/scopeResolve.cpp

```diff
@@ -1168,6 +1168,19 @@ static void setupShadowVars() {
 
 static CallExpr* resolveModuleGetNewExpr(CallExpr* call, Symbol* sym);
 
+// Track modules that we name explicitly in the scope where we do so
+static void storeReferencedMod(ModuleSymbol* mod, BaseAST* cur) {
+  if (cur == NULL) {
+    return;
+  }
+  BaseAST* scope = getScope(cur);
+  if (BlockStmt* block = toBlockStmt(scope)) {
+    block->modRefsAdd(mod);
+  } else {
+    storeReferencedMod(mod, scope);
```

Comment:
Not a big deal, but I personally think it better to write this simple traversal as a loop

Reply:
You know, I'm not entirely certain why I chose to make it a recursive function instead XD